src/main/scala/mesosphere/marathon/MarathonScheduler.scala

```diff
@@ -37,7 +38,7 @@ class MarathonScheduler @Inject() (
     frameworkId: FrameworkID,
     master: MasterInfo): Unit = {
     log.info(s"Registered as ${frameworkId.getValue} to master '${master.getId}'")
-    frameworkIdUtil.store(frameworkId)
+    Await.result(frameworkIdRepository.store(FrameworkId.fromProto(frameworkId)), zkTimeout)
```

Comment:
Looking at the underlying JNI glue code (mesos project) it appears that any runtime execption in this callback will cause the mesos driver to `abort()`. Which is probably fine ... I just don't see that fact documented anywhere except the c++ code.